### PR TITLE
[WIP] Remove most metrics global variables

### PR DESCRIPTION
**Pull request by peter-edge**
I think the usage of the metrics package may not be optimal right now, and we may want to clean this up (the only place this is used is basically in one spot in service_setup.go, and there wasn't a need for it anyway as RegisterRootScope is never called), but this is literally just wrapping Freeze, RegisterRootScope, and RootScope in a Client interface, almost nothing else. I argue this is strictly improving FX's situation heh :)

### CODE REVIEW

**Comment by peter-edge** [comment]
This actually is half-baked - we need a way to register a root scope on a Manager I think.

File: service/core.go [hunk 1]
```diff
@@ -99,6 +99,7 @@ type serviceCore struct {
 	metricsCore
 	tracerCore
 	authClient     auth.Client
+	metricsClient  metrics.Client
```
Comment (anuptalwalkar) [inline service/core.go]: Why do we need this on service core? I don't see us using metrics client being used anywhere except for initialization.

File: service/options.go [hunk 2]
```diff
@@ -47,6 +48,17 @@ func WithMetrics(scope tally.Scope, reporter tally.CachedStatsReporter) Option {
 	}
 }
 
+// WithRootScope registers the root scope for all the manager metrics
+func WithRootScope(scopeFunc metrics.ScopeFunc) Option {
```
Comment (anuptalwalkar) [inline service/options.go]: Do we need this? Whats the benefit of having RootScope available as an option on service?

**Comment by peter-edge** [comment]
Closing this for now, will revisit in a bit

**Pull request closed without merging**